apitools/org.eclipse.pde.api.tools/src/org/eclipse/pde/api/tools/internal/model/ApiBaseline.java

```diff
@@ -177,6 +179,10 @@ public class ApiBaseline extends ApiElement implements IApiBaseline, IVMInstallC
 	public ApiBaseline(String name) {
 		super(null, IApiElement.BASELINE, name);
 		fComponentsProvidingPackageCache = new ConcurrentHashMap<>(8);
+		fSystemLibraryComponentList = new CopyOnWriteArrayList<>();
+		fComponentsById = new ConcurrentHashMap<>();
+		fAllComponentsById = new ConcurrentHashMap<>();
+		fComponentsByProjectNames = new ConcurrentHashMap<>();
```

Comment:
You could already initialize these fields at their declaration. Would save the lines here and easier to determine what implementation is actually used.

Reply:
I prefer to init in constructor, this saves you many jumps while debugging.